### PR TITLE
Fix/slider default start

### PR DESCRIPTION
**Pull request by leneherng**
### Changes
1. DC Embed: fixed the colours in dark mode for Timeseries/Stacked Area chart, added `forcedTheme` to DC partials and charts to check using it instead of theme since the Embed Page is forced to use light mode: `CatalogueEmbed.theme = "light";` 
2. DC: Another issue, as the Timeseries/Stacked Area chart uses a hard-coded background colour (to maintain the translucent effect while preventing the colours from overlapping), in dark mode the png/svg download option will save the version with the dark colours on a transparent background which looks off. Not sure what's the fix for this. 
The root cause might be the translucent colours. But a Stacked Area chart without shade might be weird if we remove the shading. For now I made the bg color change when toggling theme so the issue isn't as obvious but this doesnt resolve the issue.
3. DC Index: Reduce padding between categories in mobile
4. KTM/Rapid Explorer: remove Dropdown min width requirement
5. Rapid: translate `A0: All stations`
6. Add Active Link to 404 page to get full redirect path for legacy site
7. Added Embed API to OpenDOSM for situations where filter options may change (eg. an Intraday chart), right now there is none but its here just in case, maybe can remove in future if not used
8. Income Tax: Fix hard-coded year, refactor loading state from using timeout

### CODE REVIEW

File: apps/app/pages/404.tsx [hunk 1]
```diff
@@ -1,9 +1,28 @@
...
+import { ComponentProps, useEffect, useState } from "react";
+
+const ActiveLink = ({ children, className, ...props }: ComponentProps<"a">) => {
+  const { asPath, isReady } = useRouter();
+  const [legacyPath, setLegacyPath] = useState<string>(props.href as string);
+
+  useEffect(() => {
+    // Check if the router fields are updated client-side
+    if (isReady) {
+      setLegacyPath(`https://archive.data.gov.my${asPath}`);
```
Comment (amalasyrafs) [inline apps/app/pages/404.tsx:16]: does the old datagovmy also using the same route as we do for the dataset ? eg /data-catalogue/...

Comment (leneherng) [reply apps/app/pages/404.tsx:16]: don't think they are the same, i think it uses /dataset/...

Comment (amalasyrafs) [reply apps/app/pages/404.tsx:16]: yeah what im saying is that if the redirection suppose to help point to a dataset in archieve datagovmy and its pointing to the wrong link anyway from the redirection, then wouldnt it be counter-intuitive ?

File: apps/app/dashboards/economy/income-taxation/rank-me.tsx [hunk 2]
```diff
@@ -111,7 +116,6 @@ const IncomeRank: FunctionComponent = () => {
         );
```
Comment (amalasyrafs) [inline apps/app/dashboards/economy/income-taxation/rank-me.tsx:116]: can add set loading to false in catch too.

**Pull request merged**